content/addons.md

```diff
@@ -138,20 +146,216 @@ There are several options to see the addon in action. We could use `npm link` or
+
+Similar to a normal npm package, the entry point is named `index.js`. The files exported from `addon/index.js` will be available to developers using the addon in their apps.
+
+Let's add some public methods to our addon! Don't forget to `export` your methods.
+
+```js
+// addon/index.js
+
+const moreEnthusiasm = function (phrase) {
+  return phrase + '!!!';
+}
+
+export { moreEnthusiasm }
+```
+
+Now, let's use the methods in an app:
+
+```js
+// The JavaScript file of some component in an app
+
+import Component from '@ember/component';
+import { moreEnthusiasm } from 'ember-addon-name';
+
+export default Component.extend({
+  actions: {
+    confirmExcitement() {
+      console.log(moreEnthusiasm('We made an addon'))
+      // prints "We made an addon!!!" to the console
+    }
+  }
+});
+
+```
 
-### Including stylesheets
+### Organizing public API code
 
-### Providing multiple components in one addon
+One common pattern for managing an addon's JavaScript code is to define the methods in many separate files, perhaps grouped into subfolders, import them into `index.js`, and then export them.
+
+For example, an `index.js` file might contain nothing more than imports and exports:
+
+```js
+import { moreEnthusiasm, curbedEnthusiasm } from 'our-app-name/utilities/enthusiasm.js'
+
+export { moreEnthusiasm, curbedEnthusiasm };
+```
+
+### How to keep learning
+
+This is a very tiny example of what addons can do in terms of providing JavaScript utilties to apps. For more advanced techniques, study other well established addons. Just like there are many ways and reasons to build an Ember app, the same is true for addons!
 
 ## Writing an npm package wrapper
 
+<!-- Help wanted! -->
+<!-- This section should mention when this is not necessary, and link to helpful resources like ember-auto-import -->
+
+## In-repo addons
+
+If the addon is just meant to be used in a single project, an "in-repo" addon could be created instead. The benefit is that it is lightweight, but there are some major limitations: an in-repo addon can't be shared between apps, versioned independently, or published to npm. 
+
+From within an existing Ember app, create an in-repo addon:
+
+```bash
+ember generate in-repo-addon <addon-name> [options]
+```
+
+This generates a folder called `lib/<addon-name>` that contains its own `package.json` and an `index.js` file.
+
+The most common use case for an in-repo addon is when there is a chance a component really should be a standalone library instead, but it is not yet clear if it should be broken out from the main app. Another reason is that it can be helpful for enforcing separation of concerns within an app.
```

Comment:
To me, the super-power of an "in-repo-addon" is being able to use addon APIs for adding packages, adding commands, etc to an app, even if it's not something that will be reused.